crates/adapters/okx/src/http/client.rs

```diff
@@ -1334,4 +1400,320 @@ mod tests {
         let symbols = client.get_cached_symbols();
         assert_eq!(symbols, vec![first.inst_id.to_string()]);
     }
+
+    // Test endpoint selection logic
+    mod endpoint_selection_tests {
```

Comment:
Use the standard `mod tests` unless we want to specifically compartmentalize tests (like we do for certain database tests which need to be executed sequentially).